src/makim/core.py

```diff
@@ -31,6 +34,127 @@
 from makim.scheduler import MakimScheduler
 
 
+def run_pipeline(self, pipeline_name: str) -> None:
```

Comment:
please keep the changes in the core.py file as minimum as possible.
please take a look into this PR as a reference.